presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -42,6 +42,8 @@
     private double memoryRevokingThreshold;
     private double memoryRevokingTarget;
     private boolean retryOnOutOfMemoryBroadcastJoinEnabled;
+    private boolean retryOnOutOfMemoryWithIncreasedMemorySettingsEnabled;
+    private String sparkRetryIncreasedMemorySettingsJson = "{\"sparkSettings\":{},\"prestoSettings\":{}}";
```

Comment:
Can we use `MapSplitter` to simplify the settings? One example is available at `TransactionManagerConfig`. We can separate the presto and spark settings into two configs:
- retryPrestoSessionProperties
- retrySparkSettings (or whatever name is used in spark).

In such case, we can avoid a lot of json codec tunneling

Reply:
So, basically you are saying that instead of using one config for passing both presto and spark settings, we should add two separate configs: one for passing presto session properties and other for passing spark configs. Sounds good to me. Will do the changes.